internal/client/authorization.go

```diff
@@ -98,31 +102,30 @@ func (c *accessTokenRefresher) start() {
 		}
 		expireTime, err := c.parseExpireTime()
 		if err != nil || c.shouldRefresh(expireTime) {
-			if err := c.refresh(); err != nil {
+			if err := c.refresh(configFilePath); err != nil {
 				c.log.WithError(err).Error("failed to refresh access token")
 				return
 			}
 		}
-		go c.refreshLoop(context.TODO())
+		go c.refreshLoop(context.TODO(), configFilePath)
 	})
 }
 
 var authorizer util.Singleton[accessTokenRefresher]
 
 func (c *accessTokenRefresher) accessToken() string {
-	return c.authInfo.AccessToken
+	return c.config.AuthInfo.AccessToken
 }
 
 func (c *accessTokenRefresher) rewind() {
 	c.renewals.Store(3)
 }
 
-func GetAccessToken(config *Config) string {
+func GetAccessToken(config *Config, configFilePath string) string {
 	auth := authorizer.GetOrInit(&accessTokenRefresher{
-		authInfo:           config.AuthInfo,
-		insecureSkipVerify: config.Service.InsecureSkipVerify,
+		config: config,
 	})
-	auth.start()
+	auth.start(configFilePath)
```

Comment:
To avoid race condition problems (although very unlikely)  config file path should be stored as part of accessTokenRefresher